File: src/dotnet/src/Ceataec.StandardisedVesselDataset/Validators/Errors/ValidationError.cs
namespace Ceataec.StandardisedVesselDataset.Validators.Errors;

/// <summary>
/// Represents a validation error.
/// </summary>
/// <param name="PropertyName">The property name.</param>
/// <param name="Message">The error message.</param>
public record ValidationError(string PropertyName, string Message);

File: src/cpp/src/svd/validators/errors/validation_error.cpp
#include "svd/validators/errors/validation_error.hpp"

namespace svd::validators {

ValidationError::ValidationError(std::string property_name, std::string error_message)
    : property_name_(std::move(property_name)), error_message_(std::move(error_message)) {
}

ValidationResult::ValidationResult(std::vector<ValidationError> errors)
    : errors_(std::move(errors)) {
}

} // namespace svd::validators
